### PR TITLE
Add getAdmin in action Creator,reducer,use axios #1808

### LINKED ISSUE #1808: Add getAdmin in action Creator, reducer, use axios for api
**Actual Behaviour**

Currently admin is being set only in local react state using jquery ajax method.

**Expected Behaviour**

Integrate redux for admin state, which would allow users to use admin globally. Also, use axios for api for admin

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by fragm3**
Fixes #1808 

Had to close https://github.com/fossasia/chat.susi.ai/pull/1842, as branch was old, login button was not working

Changes:

1. Created getAdmin function to fetch using axios.
2. Added getAdmin to Login.js and removed from StaticAppBar, TopBar
3. Created action creator and reducer for getAdmin
4. Added mockStore for enzyme for StaticAppBar

Demo Link:  https://pr-1877-1-fossasia-susi-web-chat.surge.sh

### CODE REVIEW

**Comment by akshatnitd** [comment]
I debugged it locally. The showAdminService API call is being cancelled. The reason seems to be the unmounting of the component, before the API call could be made. So, make a small change and call `onRequestClose` only after the Admin API has been completed.

**Comment by fragm3** [comment]
@akshatnitd I have added handleDialogClose as a callback to ensure that admin API is completed before the tab closes. About hydrating the store, should I simply use this in default store(defaultState): `isAdmin: cookies.get('isAdmin') || false`

File: src/components/Auth/Login/Login.react.js [hunk 1]
```diff
@@ -148,6 +148,27 @@ class Login extends Component {
...
+          });
+        }
+        this.setState({
+          isAdmin: showAdmin,
+        });
+      })
+      .catch(error => {
+        console.log('Failed to get admin');
+      });
+    this.handleDialogClose();
```
Comment (akshatnitd) [inline src/components/Auth/Login/Login.react.js]: Put this inside the then of the getAdmin action block.

File: src/components/Auth/Login/Login.react.js [hunk 2]
```diff
@@ -148,6 +148,27 @@ class Login extends Component {
     }
   };
 
+  checkAdmin = () => {
+    actions
```
Comment (akshatnitd) [inline src/components/Auth/Login/Login.react.js]: This should be `this.props.actions`. Please check the code before submitting.

File: src/redux/reducers/app.js [hunk 3]
```diff
@@ -27,6 +27,13 @@ export default handleActions(
         accessToken,
       };
     },
+    [actionTypes.APP_GET_ADMIN](state, { payload }) {
+      const { isAdmin } = payload;
```
Comment (akshatnitd) [inline src/redux/reducers/app.js]: The API returns `showAdmin` instead of `isAdmin`.

Comment (akshatnitd) [reply src/redux/reducers/app.js]: You can write - `const { showAdmin: isAdmin } = payload` here.

File: src/components/Auth/Login/Login.react.js (hunk 2 above)
Comment (fragm3) [reply src/components/Auth/Login/Login.react.js]: Actually actions is already defined, no need to use this.props.actions in this file

**Comment by fragm3** [comment]
The request as you said @akshatnitd of showAdmin is getting cancelled. It is not due to closing of dialog, I have tested that.

**Comment by fragm3** [comment]
@AakashMallik still showing cancelled

**Comment by AakashMallik** [comment]
Well, I'll fork your branch and look into it. I am let you know by 3am.

**Pull request closed without merging**